content/doc/developer/publishing/style-guides.adoc

```diff
@@ -0,0 +1,55 @@
+---
+title: Style Guides
+layout: developersection
+---
+
+== Plugin Naming Convention
+
+NOTE: Not all existing plugins may follow these rules, as those were established or enforced later.
+
+=== Artifact ID
+
+A plugin's artifact ID is used as the file base name and to uniquely identify the plugin in Jenkins and on update sites.
+It needs to follow some conventions to be published:
+
+* Use a lowercase ID, and separate terms with hyphens as needed.
+* Include neither +jenkins+ or +plugin+ in the ID unless necessary for the name to make any sense at all.
+
+For example, use +snapchat-notification+ instead of +JenkinsSnapchatNotificationPlugin+ or +notification-plugin-for-snapchat+.
+
+The ID cannot be changed after the first release; Jenkins would consider it a different plugin.
+
+=== Name
+
+A plugin's name is shown to users on the Jenkins UI and elsewhere, such as the plugin site.
+
+If is recommended to use a short and descriptive name.
+Including _Jenkins_ or _Plugin_ in the name is usually redundant, and these terms may be stripped from the name in some cases to ensure consistency in lists, and to shorten the name.
+
+=== Group ID
+
+We recommend use of +io.jenkins.plugins+ or +org.jenkins-ci.plugins+ group IDs, but do not prohibit other group IDs unless they're in bad faith (e.g. referencing an organization you have no relationship with).
+
+
+== Java Source Code
+
+The Jenkins project generally follows the link:www.oracle.com/technetwork/java/codeconvtoc-136057.html[Oracle Java Code Convention] but it's not well enforced even in core components.
+Individual plugin maintainers sometimes choose to use different style guides for their plugins.
+
+
+== Commit Messages
+
+Git commit messages should start with a reference to the Jira issue they're related to, if applicable, followed by a short summary on the first line, and more details on subsequent lines.
+Example:
+
+----
+[JENKINS-54321] Frobnicate the widget
+----
+
+If a given commit _fixes_ the specified issue, use of one of the following prefixes will make an automated infra process resolve the related Jira issue.
+
+----
+[FIX JENKINS-54321] Frobnicate the widget
+[FIXED JENKINS-54321] Frobnicate the widget
+[FIXES JENKINS-54321] Frobnicate the widget
+----
```

Comment:
Link to https://github.com/jenkinsci/jep/tree/master/jep/3 ?

Reply:
Seems unrelated.